### PR TITLE
Enable support toolbar for ProfileActivity and added a couple of unit…

### LINKED ISSUE #5026: Add report menu for Leaderboard
As stated at https://github.com/commons-app/apps-android-commons/pull/5025#issuecomment-1206026553 , we need to add the Report menu in Explore to Leaderboard.

### PR DESCRIPTION
**Pull request by shankarpriyank**
**Description (required)**

Fixes #5026

What changes did you make and why?
Added a couple of unit tests to #5068 

| Before | After |
| --- | --- |
| image| image

### CODE REVIEW

**Comment by nicolas-raoul** [comment]
And would you mind posting before/after screenshots? Thanks!

**Comment by shankarpriyank** [comment]
@nicolas-raoul Added the screenshots and made a very minor change in the MediaDetailPagerFragment.java.
Can you please point out some coding style violations (everything seems fine to me now)?

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java [hunk 1]
```diff
@@ -196,6 +196,10 @@ private void initWLMCampaign() {
 
     @Override
     public void onCreateOptionsMenu(@NonNull final Menu menu, @NonNull final MenuInflater inflater) {
+
+        // Removing contributions menu items for ProfileActivity
+        if(getActivity() instanceof ProfileActivity){ return;}
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java:201]: Space after if

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java [hunk 2]
```diff
@@ -641,7 +645,7 @@ public void viewPagerNotifyDataSetChanged() {
     @Override
     public void showDetail(int position, boolean isWikipediaButtonDisplayed) {
         if (mediaDetailPagerFragment == null || !mediaDetailPagerFragment.isVisible()) {
-            mediaDetailPagerFragment = new MediaDetailPagerFragment();
+            mediaDetailPagerFragment = new MediaDetailPagerFragment(false,true);
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java:648]: Space after ,

File: app/src/test/kotlin/fr/free/nrw/commons/media/MediaDetailPagerFragmentUnitTests.kt [hunk 3]
```diff
@@ -67,7 +67,7 @@ class MediaDetailPagerFragmentUnitTests {
 
         val activity = Robolectric.buildActivity(SearchActivity::class.java).create().get()
 
-        fragment = MediaDetailPagerFragment()
+        fragment = MediaDetailPagerFragment(false,true)
```
Comment (nicolas-raoul) [inline app/src/test/kotlin/fr/free/nrw/commons/media/MediaDetailPagerFragmentUnitTests.kt:70]: idem

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

**Comment by shankarpriyank** [comment]
@nicolas-raoul Thanks for pointing out the errors. I have made the changes as requested.

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java [hunk 4]
```diff
@@ -196,6 +196,10 @@ private void initWLMCampaign() {
 
     @Override
     public void onCreateOptionsMenu(@NonNull final Menu menu, @NonNull final MenuInflater inflater) {
+
+        // Removing contributions menu items for ProfileActivity
+        if (getActivity() instanceof ProfileActivity) { return;}
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java:201]: Space before `}`

File: app/src/main/java/fr/free/nrw/commons/profile/ProfileActivity.java [hunk 5]
```diff
@@ -140,14 +149,13 @@ private void setTabs() {
         fragmentList.add(leaderboardFragment);
         titleList.add(getResources().getString(R.string.leaderboard_tab_title).toUpperCase());
 
-        if (shouldShowContributions) {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/profile/ProfileActivity.java:143]: In the past, why was this `if (shouldShowContributions) {` added? You can use `git blame` to find out.

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

**Comment by shankarpriyank** [comment]
@nicolas-raoul Made the requested changes and ` if (shouldShowContributions)` was added in this pr https://github.com/commons-app/apps-android-commons/pull/4678

**Review by nicolas-raoul — APPROVED** [review]
Working well, thank you!

**Comment by shankarpriyank** [comment]
Hey @nicolas-raoul I was having a look at the issues trying to decide what to pick next. I am a bit confused right now. If you have any issues in mind, that might be suitable for me please let me know

**Pull request merged**